Update fe2 doc (#1398)
dedis/popstellar: fe2-android/docs/README.md

The application follows the Model-View-ViewModel (MVVM) pattern and uses
this [guide to app architecture](https://developer.android.com/jetpack/guide).

- The **View** consists of all the activities and fragments of the application. They are responsible for the displaying part.

> - The **ViewModel** implements the UI logic and prepares and manages the data used by the activities

Reviewer: ```suggestion
- The **ViewModel** implements the UI logic. It prepares and manages the data used by the activity
```

> - The **ViewModel** implements the UI logic. It prepares and manages the data used by the activity

Author: I prefer the current version as it makes it clearer in my opinion that a view model may not be used by several Activities

>   and fragments. Here, there's roughly a view model per feature. A fragment may use more than 1 view model.

Reviewer: ```suggestion
  and the fragments. Here, there's roughly a view model per feature. A fragment may use more than 1 view model.
```

> - The **Model** consists of the local and remote data source, the model classes and the repositories.

Reviewer: ```suggestion
- The **Model** consists of the model classes stored in the repositories.
```
Actually we should probably just dump the whole principle of remote vs local datasource

There is a whole package that is not used anywhere and it might only confuse people.

classes, but it **doesn't** modify the information contained in it. The repositories are separated roughly along feature lines, e.g. there is a `RollCallRepository` for roll call data. Repositories may only be updated in response to an incoming message from the backends.

As such, let's take the example of a user who wants to cast a vote in an election and does the necessary UI operations. In turn, the application will send a message
to the backend (which the backend should acknowledge), which will then validate it and propagate it in the system.
Eventually, the vote is sent back to the application (through the publish/subscribe channel), and upon receiving it, the application will update its state. Consequently, the `ElectionRepository` will be updated and UI elements listening to it will be updated as well.

Occasionally, the user interface could directly modify the application state, but this would only be
valid for local operations affecting local data (e.g., changing a local setting, clearing the data
stored in the browser, etc.).

As a side note, the `ViewModel` classes may act as an intermediary between the views and the repositories.

> Moreover, view models are primarily used to perform actions on behalf of the views and to store views data. To exemplify the difference between the state stored in the repositories and the state stored in view models, we will take the witness addition process. To add a witness, the main key of an LAO member will be scanned and a message will be sent to the backend(s). The `WitnessingViewModel` will update the list of scanned witness and and the number of scanned people will be displayed on the screen. On the other hand the list of witnesses (accepted and broadcast by a backend) will only be updated once the backend messages are received and processed.

Reviewer 2: ```suggestion
Moreover, view models are primarily used to perform actions on behalf of the views and to store views data. To exemplify the difference between the state stored in the repositories and the state stored in view models, we will take the witness addition process. To add a witness, the main key of an LAO member will be scanned and a message will be sent to the backend(s). The `WitnessingViewModel` will update the list of scanned witness and the number of scanned people will be displayed on the screen. On the other hand the list of witnesses (accepted and broadcast by a backend) will only be updated once the backend messages are received and processed.
```
